tests/integration_tests/performance/test_net_tput.py

```diff
@@ -0,0 +1,290 @@
+    total_sent = result['end']['sum_sent']
+    total_received = result['end']['sum_received']
+    duration = float(total_received['seconds'])
+    total_bytes_received = int(total_received['bytes'])
+    retransmits = int(total_sent['retransmits'])
+
+    # Measuring TPUT from rx-ing side.
+    tput = round((total_bytes_received*8) / (1024*1024*duration), 2)
+
+    if test not in tcp_statistics:
+        tcp_statistics[test] = create_tcp_statistics()
+
+    tcp_statistics[test]['tput'].add(tput)
+    tcp_statistics[test]['retransmits'].add(retransmits)
+
+    return {
+        'test': test,
+        'tput': tput,
+        'retransmits': retransmits,
+        'packet_size': packet_size,
+    }
+
+
+def run_udp(microvm, host_ip, results, iperfs=1):
+    """Run tests using UDP protocol."""
+    protocol = 'udp'
+    udp_statistics = {}
+
+    for _ in range(RUNS):
+        for bw in BANDWIDTHS[protocol]:
+            for mode in MODES:
+                cmd = '{} -c {} -t{} -f MBytes -J -b {}M {} {} -w {}M \
+                        -P {} -A {}'.format(
+                    IPERF,
+                    host_ip,
+                    ITERATION_DURATION,
+                    bw,
+                    MODES_OPTIONS[mode],
+                    PROTOCOL_OPTIONS[protocol],
+                    WINDOW_SIZE,
+                    iperfs,
+                    PINNING_CPU
+                )
+
+                iperf_out = spawn_guest_iperf(microvm, cmd)
+                result = parse_iperf_udp_result(iperf_out, mode,
+                                                bw, udp_statistics)
+                results.append(result)
+                print(result)
+
+    return udp_statistics
+
+
+def run_tcp(microvm, host_ip, results, iperfs=1):
+    """Run tests using TCP protocol."""
+    protocol = 'tcp'
+    tcp_statistics = {}
+
+    for _ in range(RUNS):
+        for bw in BANDWIDTHS[protocol]:
+            for mode in MODES:
+                for packet_size in PACKET_SIZE:
+                    cmd = '{} -c {} -t{} -f MBytes -J -b {}M {} {} -w {}M \
+                            -O {} -l {} -P {} -A {}'.format(
+                        IPERF,
+                        host_ip,
+                        ITERATION_DURATION,
+                        bw,
+                        MODES_OPTIONS[mode],
+                        PROTOCOL_OPTIONS[protocol],
+                        WINDOW_SIZE,
+                        OMIT_TIME,
+                        packet_size,
+                        iperfs,
+                        PINNING_CPU
+                    )
+
+                    iperf_out = spawn_guest_iperf(microvm, cmd)
+                    result = parse_iperf_tcp_result(iperf_out, mode,
+                                                    bw, packet_size,
+                                                    tcp_statistics)
+                    results.append(result)
+                    print(result)
+
+    return tcp_statistics
```

Comment:
These 2 methods look very similar. Can we have merge them somehow in order to have less duplicate code here ?

Reply:
There are some differences in terms of parameters used, but maybe it could be done by creating some kind of common ground command with the common parameters and then add specific ones in each protocol's method.